### PR TITLE
opencv overview

### PR DESCRIPTION
**Pull request by Om-Doiphode**
Converted python code snippets of opencv overview to C++.
Following code files were implemented:
1) Reading and writing images
2) Blending
3) Masking
4) Object detection
5) Contours
6) Resizing

Co authored by @vrnimje

### CODE REVIEW

File: 4_cv_basics/8_opencv_overview/masking/masking.cpp [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+using namespace std;
+
+int main()
+{
+    // Load two images
+    Mat img1 = imread("PROJECT_SOURCE_DIR/assets/images/DK.jpeg");
+    Mat img2 = imread("PROJECT_SOURCE_DIR/assets/images/bcci.png");
+    imshow("Image1", img1);
+
+    // Getting the height, width and number of channels of the image
```
Comment (Ris-Bali) [inline 4_cv_basics/8_opencv_overview/masking/masking.cpp:13]: Avoid writing comments in first_person

File: 4_cv_basics/8_opencv_overview/masking/masking.cpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    // I want to put logo on top - left corner, So I create a ROI
```
Comment (Ris-Bali) [inline 4_cv_basics/8_opencv_overview/masking/masking.cpp:14]: Same case here

File: 4_cv_basics/8_opencv_overview/masking/masking.cpp [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    int rows = img2.rows, cols = img2.cols, channels = img2.channels();
+
+    // Cropping the img1 with (0,0) as the top left corner and (rows,cols) as the bottom right corner
```
Comment (Ris-Bali) [inline 4_cv_basics/8_opencv_overview/masking/masking.cpp:17]: Write more concise comments . You can explain this in the README

**Comment by Ris-Bali** [comment]
Is this pr ready for review ?

**Comment by vrnimje** [comment]
Yes @Ris-Bali

File: 4_cv_basics/8_opencv_overview/masking/masking.cpp [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    // inverting colors
+    Mat mask_inv;
+    bitwise_not(mask, mask_inv);
+
+    // Resizing mask
+    resize(mask, mask, roi.size());
+
+    // Blacking-out the area of logo in ROI
+    Mat img1_bg;
+    bitwise_and(roi, roi, img1_bg, mask);
```
Comment (Ris-Bali) [inline 4_cv_basics/8_opencv_overview/masking/masking.cpp:39]: Add a readme expalining what masking is and some images

File: 4_cv_basics/8_opencv_overview/contours/contours.cpp [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    Mat contourImg = Mat::zeros(img.size(), CV_8UC3);
+    for (size_t i = 0; i < contours.size(); i++)
+    {
+        drawContours(img, contours, i, Scalar(0, 255, 0), 2);
+    }
+
+    imshow("Contours", img);
+    waitKey(0);
+    destroyAllWindows();
+}
```
Comment (Ris-Bali) [inline 4_cv_basics/8_opencv_overview/contours/contours.cpp:37]: Add a readme for contours too

File: 4_cv_basics/8_opencv_overview/blending/blending.cpp [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+    Mat res;
+
+    float alpha = 0.6, beta = 1 - alpha;
+
+    addWeighted(image1, alpha, image2, beta, 0.0, res);
+
+    namedWindow("Display Image");
+    imshow("Display Image", res);
+    waitKey(0);
```
Comment (Ris-Bali) [inline 4_cv_basics/8_opencv_overview/blending/blending.cpp:22]: Can you add a comment here to state what is addweighted doing (i.e. blends img1 and img2).

File: 4_cv_basics/8_opencv_overview/README.md [hunk 7]
```diff
@@ -0,0 +1,725 @@
...
+## **Applications of Computer Vision :**
+
+1. **Medicine Application** :
+
+- Classification and detection (e.g. lesion or cells classification and tumor detection)
+- 3D human organ reconstruction (MRI or ultrasound)
+- Vision-guided robotics surgery
+
+---
+
```
Comment (Ris-Bali) [inline 4_cv_basics/8_opencv_overview/README.md:35]: Can you divide this readme (i.e. separate readme for each topic)

File: 4_cv_basics/8_opencv_overview/README.md [hunk 8]
```diff
@@ -0,0 +1,725 @@
...
+### Countours
+
+Contours can be explained simply as a curve joining all the continuous points (along the boundary), having same color or intensity. The contours are a useful tool for shape analysis and object detection and recognition.
+
+Illustration:
+
+<img src="../../assets/images/contours_1.png" />
+
+## Expected Results
```
Comment (amanchhaparia) [inline 4_cv_basics/8_opencv_overview/README.md:562]: Place the content of `Expected Results` in their specific  topic sections

File: 4_cv_basics/8_opencv_overview/README.md [hunk 9]
```diff
@@ -0,0 +1,725 @@
...
+<th style="text-align:center" colspan="2"> Contours</th>
+</tr>
+<tr>
+<td style="text-align: center" colspan="2"><image src="../../assets/images/contours.png" alt = "Image1" width = 350 height="200"></td>
+</table>
+
+
+<table>
+<tr>
+<th colspan="2" style="text-align:center"> MORPHOLOGICAL OPERATIONS </th>
```
Comment (amanchhaparia) [inline 4_cv_basics/8_opencv_overview/README.md:669]: Create a section for Morphological Operations as well.

File: 4_cv_basics/8_opencv_overview/grayscale/grayscale.cpp [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+    auto x = im2.size;
+
+    cout << x[0] << ',' << x[1] << endl;
+    cout << im2.channels() << endl;
+
+    namedWindow("Display Image");
+    imshow("Display Image", im2);
+    waitKey(0);
+
+    return 0;
```
Comment (amanchhaparia) [inline 4_cv_basics/8_opencv_overview/grayscale/grayscale.cpp:29]: Add comments

**Review by amanchhaparia — CHANGES_REQUESTED** [review]
- Add a `table of content` in the readme.
- Make a consistent use of `OpenCV`, instead of OpenCv/Open CV.
- Add a new line in files where required.

**Review by amanchhaparia — CHANGES_REQUESTED** [review]
@Om-Doiphode, What is the need of the blob detection jupyter notebook in this PR?

**Review by amanchhaparia — CHANGES_REQUESTED** [review]
- Revert the changes made in root Makefile
- Move the assets folder under the opencv_overview folder instead
- Add `table of content` in the root readme of opencv_overview folder.
- Move the long table of the functions and its description at the end or divide in another readme

**Review by amanchhaparia — CHANGES_REQUESTED** [review]
- No need to modify the root `Makefile`.
- The scope of the PR should be restricted to `opencv_overview` folder. Do not make unnecessary changes with rest of the folder.

**Comment by amanchhaparia** [comment]
@Om-Doiphode, why you are unnecessary changing the contents of the other folders. Kindly restrict the commits of this PR to the opencv folder only

**Pull request closed without merging**